scripts/forc-build-with-typegen.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+set -o pipefail
+
+output=$(pnpm fuels-forc build -p test/fixtures/forc-projects --release | tee >(cat >&2))
+
+exit_code=$?
+
+if [ $exit_code -ne 0 ]; then
+    exit $exit_code
+fi
+
+res=$(echo "$output" | grep Compiling | grep -v library | sed -E 's/.* (predicate|contract|script).*\((.*)\)/\1 -i \2\/out\/release\/*-abi.json/g')
```

Comment:
@nedsalk At this point, we can dogfood one level deeper, and `pnpm fuels build` the whole thing directly. However, since the built-in binaries are being [removed](https://github.com/FuelLabs/fuels-ts/issues/2084#issuecomment-2072178268) in #2143, we'd need a tiny [new trick](https://github.com/FuelLabs/fuels-ts/pull/2143/files#r1576798648) to tell `fuels` to use our `@internal/{forc,fuel-core}` binaries. (cc @petertonysmith94)

Reply:
I like the `pnpm fuels build` idea. There's a lot going on with this switch to `fuelup` so I'll put this PR into draft mode until that is resolved.

Reply:
After a chat with @danielbate and his explanation in #2312 
> Uses fuels to build projects, means we don't have to be blocked by https://github.com/FuelLabs/fuels-ts/pull/2143 as the underlying binaries can be swapped without impacting the command entrypoint, so no need for additional scripts

I moved to `pnpm fuels build` easily.